extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -14,5 +14,88 @@
  * limitations under the License.
  */
 
+import {Messaging} from './messaging.js';
+import {listen} from '../../../src/event-helper';
+import {viewerForDoc} from '../../../src/viewer';
+import {dev,user} from '../../../src/log';
+
+
+/**
+ * @fileoverview This is the communication protocol between AMP and the viewer.
+ * This should be included in an AMP html file to communicate with the viewer.
+ */
 export class AmpViewerIntegration {
+
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @const @private {!Window} win */
+    this.win_ = win;
+  }
+
+  /**
+   * Initiate the handshake. If handshake confirmed, start listening for
+   * messages.
+   * @return {!Promise|undefined}
+   */
+  init() {
+    const isIframed = window.parent && window.parent != window;
+    if (!isIframed) {
+      return;
+    }
+    dev().info('AVI', 'amp-viewer-integration.js => handshake init()');
```

Comment:
Create a const up top `const TAG = 'amp-viewer-integration'` and use it in its stead.